app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumberForm.kt

```diff
@@ -40,22 +41,45 @@ class AddHousenumberForm : AbstractQuestFormAnswerFragment<HousenumberAnswer>()
     private var addButton: View? = null
     private var subtractButton: View? = null
 
-    private var isHousename: Boolean = false
+    enum class IntefaceMode {
+        HOUSENUMBER, HOUSENAME, HOUSENUMBER_AND_HOUSENAME
+    }
+    private var interfaceMode: IntefaceMode = IntefaceMode.HOUSENUMBER
 
     private var houseNumberInputTextColors: ColorStateList? = null
 
     private val isShowingHouseNumberHint: Boolean get() = houseNumberInputTextColors != null
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         val view = super.onCreateView(inflater, container, savedInstanceState)
-        isHousename = savedInstanceState?.getBoolean(IS_HOUSENAME) ?: false
-        setLayout(if(isHousename) R.layout.quest_housename else R.layout.quest_housenumber)
+        val isHousename = savedInstanceState?.getBoolean(IS_HOUSENAME) ?: false
+        val isHousenumber = savedInstanceState?.getBoolean(IS_HOUSENUMBER) ?: false
+        interfaceMode = when {
+            isHousename && isHousenumber -> IntefaceMode.HOUSENUMBER_AND_HOUSENAME
+            isHousename -> IntefaceMode.HOUSENAME
+            isHousenumber -> IntefaceMode.HOUSENUMBER
+            else -> IntefaceMode.HOUSENUMBER
+        }
+        setLayout(
+            when(interfaceMode) {
+                IntefaceMode.HOUSENUMBER -> R.layout.quest_housenumber
+                IntefaceMode.HOUSENAME -> R.layout.quest_housename
+                IntefaceMode.HOUSENUMBER_AND_HOUSENAME -> R.layout.quest_housename_and_housenumber
+            }
+        )
         return view
     }
 
     override fun onSaveInstanceState(outState: Bundle) {
         super.onSaveInstanceState(outState)
-        outState.putBoolean(IS_HOUSENAME, isHousename)
+        when(interfaceMode) {
+            IntefaceMode.HOUSENUMBER -> outState.putBoolean(IS_HOUSENUMBER, true)
+            IntefaceMode.HOUSENAME -> outState.putBoolean(IS_HOUSENAME, true)
+            IntefaceMode.HOUSENUMBER_AND_HOUSENAME -> {
+                outState.putBoolean(IS_HOUSENUMBER, true)
+                outState.putBoolean(IS_HOUSENAME, true)
+            }
+        }
```

Comment:
```suggestion
        outState.putString(INTERFACE_MODE, interfaceMode.name)
```

and in `onCreateView`
```kotlin
interfaceMode = savedInstanceState?.getString(INTERFACE_MODE, null)?.let { InterfaceMode.valueOf(it) }
```
(or)
```kotlin
interfaceMode = if (savedInstanceState != null) {
  InterfaceMode.valueOf(savedInstanceState.getString(INTERFACE_MODE, null)!!)
} else InterfaceMode.HOUSENUMBER
```

Reply:
I think the first option is missing a `?: InterfaceMode.HOUSENUMBER`. I was playing around with it and think I like
```kt
val prevMode = savedInstanceState?.getInt(INTERFACE_MODE)?.let { InterfaceMode.valueOf(it) }
interfaceMode = prevMode ?: InterfaceMode.HOUSENUMBER
```

Reply:
You are right

Reply:
You can stay under the 100 char limit using a function reference :laughing: 
```kt
        val prevMode = savedInstanceState?.getString(INTERFACE_MODE)?.let(InterfaceMode::valueOf)
```